### PR TITLE
Adding Inline example for psislw and r2_score.

### PR DESCRIPTION
**Pull request by smit-s**
## Description

psislw and r2_score functions in stats.py did not have any inline examples. This PR adds inline example to these methods and addresses issue  #1057.

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/stats/stats.py [hunk 1]
```diff
@@ -792,6 +804,18 @@ def r2_score(y_true, y_pred):
...
     r2_std: standard deviation of the Bayesian R².
+
+    Examples
+    --------
+    Calculate R² for Bayesian regression models :
+
+    .. ipython::
+
+        In [1]: import arviz as az
+           ...: y_true=np.random.randn(10)
```
Comment (aloctavodia) [inline arviz/stats/stats.py:815]: maybe we can use this example

data = az.load_arviz_data('regression1d')
y_true = data.observed_data["y"].values
y_pred = data.posterior_predictive.stack(sample=("chain", "draw"))['y'].values.T
az.r2_score(y_true, y_pred)

Comment (smit-s) [reply arviz/stats/stats.py:815]: Yes sure. I will make changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +17,7 @@
 ### Documentation
 * Image thumbs generation updated to be Bokeh 2 compatible (#1116)
 * Add new examples for `plot_pair` (#1110)
-
+* Add examples for `psiswl` and `r2_score` (#1129)
```
Comment (OriolAbril) [inline CHANGELOG.md:20]: psislw, the w and l are inverted

File: arviz/stats/stats.py [hunk 3]
```diff
@@ -611,6 +611,18 @@ def psislw(log_weights, reff=1.0):
         Smoothed log weights
     kss : array
         Pareto tail indices
+
+    Examples
+    --------
+    Get Pareto smoothed importance sampling (PSIS):
```
Comment (OriolAbril) [inline arviz/stats/stats.py:617]: Get Pareto smoothed importance sampling (PSIS) log weights.

We should also add a reference, i.e. https://arxiv.org/abs/1507.02646 (see `ess` function for an example on the References section)

File: CHANGELOG.md (hunk 2 above)
Comment (smit-s) [reply CHANGELOG.md:20]: Sorry for this. It was very careless of me. I'll rectify this.

Comment (OriolAbril) [reply CHANGELOG.md:20]: Don't worry, we all make typos :)

**Comment by ahartikainen** [comment]
Looks good, after addressing the reference issue, this is ready.

**Pull request merged**